active/003-seq-feature.md

```diff
@@ -0,0 +1,121 @@
+---
+Feature name: sequence-positional-metadata
+Start date: <2015-12-18>
+Pull request: 1
+Authors: ["@RNAer", "@mortonjt"]
+Contributors:
+- "@RNAer"
+- "@mortonjt"
+---
+
+# Summary
+
+Modify the `Sequence` class to incorporate the new pandas `IntervalIndex` into `positional_metadata`.
+
+# Motivation
+
+The `IntervalIndex` is not merged into pandas yet. To test the code below,
+you will need the specific branch of `pandas` from github:
+
+```
+git clone git@github.com:shoyer/pandas.git
+cd pandas
+git pull origin IntervalIndex
+pip install -e .
+```
+
+The current implementation of `positional_metadata` in the `Sequence` class takes > 1TB
+memory to read in a E. coli genome with 260K features. And the `pandas` sparse dataframe
+can alleviate the mem usage, but it is still too memory hogging. The dicussion is mainly
+in this [thread](https://github.com/biocore/scikit-bio/issues/1159).
+
+Incorporating an IntervalIndex would also enable fast query times for fetch intervals,
+which is not supported by the current prototype of the `positional_metadata` at the moment.
+For example:
+
+```python
+from pandas.core.interval import Interval, IntervalIndex
+intervals = [(190, 255),
+             (190, 255),
+             (250, 2799),
+             (337, 2799),
+             (2801, 3733)]
+iid = IntervalIndex.from_tuples(intervals)
+df = pd.DataFrame(data={'feature': ['gene1', 'gene2', 'gene3', 'gene4', 'gene5']},
+                  index=iid)
+df.loc[254, :]
+```
+Because its implementation of `IntervalTree`, it will return all the features containing
+coordinate of 254 very fast:
+```
+            feature
+(190, 255]    gene1
+(190, 255]    gene2
+(250, 2799]   gene3
+```
+
+# Detailed design
+
+Use `IntervalIndex` in `pandas`.  The IntervalIndex is functional for our purposes, and we don't have to even maintain it.
+
+The `positional_metadata` dataframe has an IntervalIndex as its row index and has feature metadata as column(s) so that each row is sequence region and a feature can have multiple rows/regions. The `positional_metadata` can have duplicate row index to accommandate the scenario that a single region belongs to multiple features.
+
+The dataframe should have two columns, one for the name of the feature so we can fetch the feature by name of `str` easily, the other for other metadata of the features. It would be argued that the feature medatadata column contains a hashable, dict-like object. Making it hashable will be convenient for some `pandas` operations. (The hashable implementation is [here](https://github.com/biocore/scikit-bio/pull/1157/files).
+
+
+# Drawbacks
+
+1. The `IntervalIndex` is still in active development and not merged into offical branch.
```

Comment:
Can you move drawback 1 into the detailed design section and rephrase? It's necessary for `IntervalIndex` to be in a pandas release before adding this functionality to scikit-bio so I see it as one of the implementation steps.